src/main/java/com/java_essential/varycheva/essential/homework/hw9/task3/Device.java

```diff
@@ -0,0 +1,49 @@
+package com.java_essential.varycheva.essential.homework.hw9.task3;
+
+import java.util.Objects;
+
+public class Device {
+    private String manufacturer;
+    private float price;
+    private String serialNumber;
+
+    public Device(String manufacturer, float price, String serialNumber) {
+        this.manufacturer = manufacturer;
+        this.price = price;
+        this.serialNumber = serialNumber;
+    }
+
+    public String getManufacturer() {
+        return manufacturer;
+    }
+
+    public float getPrice() {
+        return price;
+    }
+
+    public String getSerialNumber() {
+        return serialNumber;
+    }
+
+    @Override
+    public String toString() {
+        return "Device: " +  "manufacturer='" + manufacturer + '\'' +
+                ", price=" + price +
+                ", serialNumber='" + serialNumber;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Device device = (Device) o;
+        return Float.compare(device.price, price) == 0 &&
+                Objects.equals(manufacturer, device.manufacturer) &&
+                Objects.equals(serialNumber, device.serialNumber);
+    }
```

Comment:
Avoid bad readable code, simplify to variant bellow
```suggestion
   @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        
        final Device other = (Device) obj;
        
        if (!Objects.equals(this.manufacturer, other.manufacturer)) {
            return false;
        }
        
        if (!Objects.equals(this.price, other.price)) {
            return false;
        }

        if (!Objects.equals(this.serialNumber, other.serialNumber)) {
            return false;
        }
        
        return true;
    }
```
Main rule it's readable and fast extending friendly code!